File: spring-boot-commons-util/src/test/java/com/github/springframework/boot/commons/util/StringsTest.java
package com.github.springframework.boot.commons.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class StringsTest {

	@Test
	void testNewInstance() {
		UnsupportedOperationException e = assertThrows(UnsupportedOperationException.class, Strings::new);
		assertEquals("Utility class should not be instantiated", e.getMessage());
	}

	@Test
	void testDefaultIfNull() {
		assertEquals(Strings.EMPTY, Strings.defaultIfNull(null, Strings.EMPTY));
		assertEquals("testValue", Strings.defaultIfNull("testValue", "testDefaultValue"));
	}

	@Test
	void testIsEmailAddress() {
		assertFalse(Strings.isEmailAddress(null));
		assertFalse(Strings.isEmailAddress("test"));
		assertTrue(Strings.isEmailAddress("test@test.com"));
	}

}

File: spring-boot-commons-base/src/test/java/com/github/awesome/springboot/commons/base/StringsTest.kt
package com.github.awesome.springboot.commons.base;

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import kotlin.test.assertEquals

class StringsTest {

    @Test
    fun testNewInstance() {
        val e = assertThrows<UnsupportedOperationException> {
            Strings()
        }
        assertEquals("Utility class should not be instantiated", e.message)
    }

    @Test
    fun testDefaultIfNullOrEmpty() {
        assertEquals("default", Strings.defaultIfNullOrEmpty(null, "default"))
        assertEquals("default", Strings.defaultIfNullOrEmpty(Strings.EMPTY, "default"))
        assertEquals("value", Strings.defaultIfNullOrEmpty("value", "default"))
    }

}
